src/serac/physics/heat_transfer.hpp

```diff
@@ -328,17 +380,36 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
   /**
    * @brief Set the thermal flux boundary condition
    *
-   * @tparam FluxType The type of the flux function
-   * @param flux_function A function describing the thermal flux applied to a boundary
+   * @tparam FluxType The type of the thermal flux object
+   * @param flux_function A function describing the flux applied to a boundary
+   *
+   * @pre FluxType must be a object that can be called with the following arguments:
+   *    1. `tensor<T,dim> x` the spatial coordinates for the quadrature point
+   *    2. `tensor<T,dim> n` the outward-facing unit normal for the quadrature point
+   *    3. `double t` the time (note: time will be handled differently in the future)
+   *    4. `T temperature` the current temperature at the quadrature point
+   *    4. `tuple{value, derivative}`, a variadic list of tuples (each with a values and derivative),
+   *            one tuple for each of the trial spaces specified in the `DependsOn<...>` argument.
+   *
+   * @note The actual types of these arguments passed will be `double`, `tensor<double, ... >` or tuples thereof
+   *    when doing direct evaluation. When differentiating with respect to one of the inputs, its stored
+   *    values will change to `dual` numbers rather than `double`. (e.g. `tensor<double,3>` becomes `tensor<dual<...>,
+   * 3>`)
    *
-   * @pre FluxType must have the operator (x, normal, temperature) to return the thermal flux value
+   * @note: until mfem::GetFaceGeometricFactors implements their JACOBIANS option,
+   * (or we implement a replacement kernel ourselves) we are not able to compute
+   * shape sensitivities for boundary integrals.
    */
   template <int... active_parameters, typename FluxType>
   void setFluxBCs(DependsOn<active_parameters...>, FluxType flux_function)
   {
-    K_functional_->AddBoundaryIntegral(
-        Dimension<dim - 1>{}, DependsOn<0, active_parameters + 1 ...>{},
-        [flux_function](auto x, auto n, auto u, auto... params) { return flux_function(x, n, u, params...); }, mesh_);
+    residual_->AddBoundaryIntegral(
+        Dimension<dim - 1>{}, DependsOn<0, 1, 2, active_parameters + NUM_STATE_VARS...>{},
+        [this, flux_function](auto x, auto n, auto u, auto /* dtemp_dt */, auto shape, auto... params) {
+          auto p = get<VALUE>(shape);
+          return flux_function(x + p, n, ode_time_point_, u, params...);
```

Comment:
It's slightly suspicious that there is a minus sign on the heat source but not on the surface flux, but that will all come out in the wash once there is a patch test.

Reply:
I'll do a quick sanity check to verify.

Reply:
Yep - this sign is wrong. Fix and updated test incoming...